packages/trimerge-sync/src/TrimergeClient.ts

```diff
@@ -330,66 +342,81 @@ export class TrimergeClient<
   ): void {
     const { ref, baseRef, mergeRef } = asCommitRefs(commit);
     if (this.commits.has(ref)) {
-      // Promote lazy commit
+      // Promote temp commit
       if (type === 'external') {
-        this.lazyCommits.delete(ref);
+        this.promoteTempCommit(ref);
```

Comment:
So this would happen in the case of temp merge that gets confirmed by the server, right? do we need to do any sort of confirmation here where we update the commit with serverInfo?

Reply:
yea, we need to handle that upsert-like behavior I think once we figure out the metadata buckets